### PR TITLE
Add Saddle Console

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds a console to make it easy to quickly pull into a JavaScript REPL and communicate with a saddle deployment. We allow relatively simple deployments, access to `saddle` and deployed contracts, and start to build out a diverse set of functionality which can make a saddle console your goto way of simple contract interactions.

```bash
npx -n --experimental-repl-await saddle console
```

```javascript
Saddle console on network development Web3ProviderEngine

> saddle
Saddle {
  ...
}

> .deploy Counter
Deploying contract Counter with args []
Deployed Counter at 0x81d2b78e483Ad6e9bc8e1a46F45434cFbad980B5
> await counter.methods.count().call()
0
```

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -44,18 +45,30 @@ if (require.main === module) {
...
+      startConsole(argv.network, argv.trace, argv.verbose);
+    })
     .command('deploy <contract>', 'Deploy a contract to given network', (yargs) => {
       return yargs
         .positional('contract', {
           describe: 'Contract to deploy (e.g. myContract.sol)',
           type: 'string'
         });
     }, (argv) => {
+      console.log("hereee");
```
Comment (mykelp) [inline src/cli.ts]: Does this need to be `hereee`?

File: README.md [hunk 2]
```diff
@@ -150,16 +150,35 @@ If you have new features or analysis you'd like to add to tracing, feel free to
...
+npx -n --experimental-repl-awaitsaddle console
+```
+
+```javascript
+Saddle console on network development Web3ProviderEngine
+> saddle
+Saddle {
+  ...
+}
+> .deploy Counter
```
Comment (mykelp) [inline README.md:175]: Is the command `.deploy` or `deploy`?

**Review by jflatow — APPROVED** [review]
🆒 😎

File: README.md (hunk 2 above)
Comment (hayesgm) [reply README.md:175]: .deploy, it's a `dot` command. you could also do `saddle.deployContract(...)` if you want to spell it out in code

**Pull request merged**